### PR TITLE
Add basic api 'get' endpoint and method to return vendor codes

### PR DESCRIPTION
**Pull request by mattlindsey**
Add 'get' endpoint to pass in the 'redeemify' code and get back one or more vendor codes.

This doesn't yet have authorization built in, but for the sake of having small PRs, I would like to merge this.

### CODE REVIEW

File: app/controllers/redeemify_codes_controller.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  before_action :set_code, only: [:show]
+
+  def show
+    render json: @code
+  end
+
+  private
+
+  def set_code
+    if current_user then
```
Comment (tansaku) [inline app/controllers/redeemify_codes_controller.rb]: do we need the `then`?  

should we prefer to check for current_user using 

```
before_action :authenticate_user!
```

File: app/models/redeemify_code.rb [hunk 2]
```diff
@@ -1,4 +1,8 @@
 class RedeemifyCode < ActiveRecord::Base
 	belongs_to :provider
 	belongs_to :user
+  has_many :vendor_codes
```
Comment (tansaku) [inline app/models/redeemify_code.rb]: inconsistent indentation?

File: spec/controllers/users_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
```
Comment (tansaku) [inline spec/controllers/users_controller_spec.rb]: I would delete this - controller specs serve no purpose IMHO

File: spec/requests/api/v1/api_spec.rb [hunk 4]
```diff
@@ -0,0 +1,21 @@
+require 'rails_helper'
+require 'spec_helper'
+
+RSpec.describe RedeemifyCodesController, type: :request do
+  describe "GET /users/:user_id/redeemify_codes/:code" do
```
Comment (tansaku) [inline spec/requests/api/v1/api_spec.rb]: looks find, but we could move the get into a subject and then create separate `it` statements for each assertion

**Comment by tansaku** [comment]
good start - any desire to add cukes?

File: app/controllers/redeemify_codes_controller.rb (hunk 1 above)
Comment (mattlindsey) [inline app/controllers/redeemify_codes_controller.rb]: I'm going to be writing 'authenticate user' in the next ticket, so if it's OK with you I'll just leave as is for now.

File: spec/requests/api/v1/api_spec.rb (hunk 4 above)
Comment (tansaku) [inline spec/requests/api/v1/api_spec.rb]: great!  We could also use named subject, e.g. 

```rb
subject(:make_get_request) { get '/redeemify_codes/:id', format: :json, params: { id: @rcode.code } }
```

see https://www.relishapp.com/rspec/rspec-core/docs/subject/explicit-subject

File: app/controllers/redeemify_codes_controller.rb [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+  before_action :set_code, only: [:show]
+
+  def show
+    render json: @code
+  end
+
+  private
+
+  def set_code
+    if current_user
```
Comment (mattlindsey) [inline app/controllers/redeemify_codes_controller.rb]: Sorry.  You're comment and my response got erased by my last commit!
I replied that I am using 'before_action', so can we wait until I do the auth stuff to do a authenticate_user method?

Comment (tansaku) [inline app/controllers/redeemify_codes_controller.rb]: guess so - but couldn't we have 

```rb
 before_action :set_code, only: [:show], if: :current_user
```

File: config/routes.rb [hunk 6]
```diff
@@ -31,13 +32,15 @@
   match "/auth/failure", to: "sessions#failure", via: [:get, :post]
   match "/logout", to: "sessions#destroy", :as => "logout", via: [:get, :post]
   match "/logout", to: "vendors#destroy", :as => "logout2", via: [:get, :post]
+
   resources :sessions
+
+  #match "/users/:user_id/redeemify_codes/:code", to: "redeemify_codes#show", via: [:get]
```
Comment (tansaku) [inline config/routes.rb]: we should delete comments right? :-)

**Comment by tansaku** [comment]
okay, so this is good to go if we are happy to proceed without cukes - any progress on the travis issue?

**Pull request merged**